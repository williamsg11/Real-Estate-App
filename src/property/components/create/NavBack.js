import React, {Component, PropTypes} from 'react';
import {StyleSheet, Text, TouchableHighlight, View} from 'react-native';
import colors from '../../../common/colors';
import Ionicons from 'react-native-vector-icons/Ionicons';
import isNull from 'lodash/isNull';

export default class NavBack extends Component {
  static propTypes = {
    stage: PropTypes.number.isRequired,
    icon: PropTypes.string,
    text: PropTypes.string,
  };

  goBack() {
    this.props.onPress();
  }

  render() {
    const {style, text, icon, stage} = this.props;
    return (
      <View style={{flex: 1}}>
        {!isNull(stage) &&
          stage > 1 &&
          <TouchableHighlight
            style={styles.container}
            onPress={() => {
              this.goBack();
            }}
            underlayColor="transparent">
            {icon
              ? <Ionicons
                  name={icon}
                  size={33}
                  color={colors.accent}
                  style={[styles.icon]}
                />
              : <Text style={[styles.title, style]}>{text}</Text>}
          </TouchableHighlight>}
      </View>
    );
  }
}

NavBack.propTypes = {
  text: React.PropTypes.string,
  icon: React.PropTypes.string,
  stage: React.PropTypes.number,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 8,
    paddingRight: 10,
  },
  title: {
    color: colors.accent,
    fontSize: 15,
  },
  icon: {
    width: 13,
    height: 33,
    alignSelf: 'center',
  },
});
