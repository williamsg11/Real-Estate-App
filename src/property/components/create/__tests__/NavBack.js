import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavBack from './../NavBack';

test('renders NavBack', () => {
  const tree = renderer.create(<NavBack />).toJSON();
  expect(tree).toMatchSnapshot();
});
