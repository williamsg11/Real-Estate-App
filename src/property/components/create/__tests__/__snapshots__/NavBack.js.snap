// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavBack 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
